### PR TITLE
feat: add support for `VocabProperty`

### PR DESCRIPTION
**Pull request by bobeal**

### CODE REVIEW

File: search-service/src/test/resources/ngsild/expectations/beehive_vocab_property_temporal_values.jsonld [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    {
+                      "@value": "Ma belle ruche",
+                      "@language": "fr"
+                    }
+                  ]
+                },
+                {
+                  "@value": "2020-03-25T08:33:17.965206Z"
+                }
+              ]
```
Comment (ranim-n) [inline search-service/src/test/resources/ngsild/expectations/beehive_vocab_property_temporal_values.jsonld:43]: only `LanguageProperty` is used here , there are no `VocabProperty`

File: search-service/src/test/resources/ngsild/fragments/temporal_instance_vocab_fragment.jsonld [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+    "type": "LanguageProperty",
+    "languageMap": {
+        "fr": "Ma belle ruche",
+        "it": "Il mio bellissimo alveare"
+    },
+    "observedAt": "2023-03-13T12:33:06Z"
+}
```
Comment (ranim-n) [inline search-service/src/test/resources/ngsild/fragments/temporal_instance_vocab_fragment.jsonld:5]: in this fragment also

File: search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt [hunk 3]
```diff
@@ -874,6 +875,12 @@ class TemporalEntityAttributeService(
                     null,
                     null
                 )
+            TemporalEntityAttribute.AttributeType.VocabProperty ->
+                Triple(
+                    serializeObject(attributePayload.getMemberValue(NGSILD_VOCABPROPERTY_VALUE)!!),
+                    null,
+                    null
+                )
```
[the remark this answers is not in the document]
Comment (bobeal) [reply search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt:883]: Yes. The _String_ is the serialized representation of anything that is not a double, nor a coordinate.

File: search-service/src/test/kotlin/com/egm/stellio/search/util/PatchAttributeTests.kt [hunk 4]
```diff
@@ -83,6 +83,32 @@ class PatchAttributeTests {
...
+                        }
+                    }
+                    """.trimIndent(),
+                    """
+                    {
+                        "attribute": {
+                            "type": "VocabProperty",
+                            "vocab": "egm"
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (bobeal) [reply search-service/src/test/kotlin/com/egm/stellio/search/util/PatchAttributeTests.kt:110]: There are 3 things in each series: current value, update payload, expected result.

File: search-service/src/test/resources/ngsild/expectations/beehive_vocab_property_temporal_values.jsonld (hunk 1 above)
Comment (bobeal) [reply search-service/src/test/resources/ngsild/expectations/beehive_vocab_property_temporal_values.jsonld:43]: discussed yesterday, it is actually a `VocabProperty` :)

**Review by ranim-n — APPROVED** [review]

**Pull request merged**